node/node.go

```diff
@@ -268,6 +272,17 @@ func MakeFull(log logging.Logger, rootDir string, cfg config.Local, phonebookAdd
 		}
 	}
 
+	if cfg.TraceServer == "" {
+		node.tracer = &graphtrace.NopClientSingleton
+	} else {
+		node.tracer, err = graphtrace.NewTcpClient(cfg.TraceServer)
+		if err != nil {
+			log.Errorf("unable to create trace client: %v", err)
+			return nil, err
+		}
+		gossip.Trace = node.tracer
+	}
+
```

Comment:
It doesn't seems that you really need it here. instead, just pass the cfg.TraceServer to 
gossip.InitializeNetworkMessageTracer(..).
The InitializeNetworkMessageTracer would be a function pointer which (by default) would be an empty function, and with the build tag would contain the above initialization.

Same goes for the usage of the tracer.. which would allow you to compile it with and without via a build tag.